index.js

```diff
@@ -437,7 +437,7 @@ var TransitionGroup = function (_React$Component) {
       if (currentChildMapping && currentChildMapping.hasOwnProperty(key)) {
         // This entered again before it fully left. Add it again.
         _this.performEnter(key);
-      } else {
+      } else if (component) {
```

Comment:
this `component` isn't the TransitionGroup instance. The error would be related to TransitionGroup not being mounted when setState is called, it'd be better to check if the current component is mounted